Source/ACE/Entity/Landblock.cs

```diff
@@ -303,6 +308,45 @@ public List<WorldObject> GetWorldObjectsByGuid(ObjectGuid objectguid, bool neigh
             }
             return allworldobj;
         }
+        /// <summary>
+        /// This is going to find a single object by guid and search both the current landblock and all adjacent landblocks.
+        /// If more than one item is found - this should not happen, it is going to return a null and punt. Og II
+        /// </summary>
+        /// <param name="objectGuid"></param>
+        /// <returns></returns>
+        public WorldObject GetWorldObjectByGuid(ObjectGuid objectGuid)
+        {
+            var objectList = GetWorldObjectsByGuid(objectGuid, true);
+
+            // if we get anything other than 1 punt.
+            if (objectList != null && objectList.Count == 1) return objectList[0];
+            return null;
+        }
+
+        /// <summary>
+        /// Check to see if we are close enough to interact with the target object.  If we are not, then save off the action, return false
+        /// if we pass the range check return true. We are "go" for action.
+        /// </summary>
+        /// <param name="player"></param>
+        /// <param name="targetWorldObject"></param>
+        /// <param name="action"></param>
+        /// <param name="movementType"></param>
+        /// <returns></returns>
+        public bool InRangeForAction(Player player, WorldObject targetWorldObject, QueuedGameAction action, MovementTypes movementType)
+        {
+            var csetup = SetupModel.ReadFromDat(targetWorldObject.PhysicsData.CSetup);
+            var arrivedRadius = (float)Math.Pow((targetWorldObject.GameData.UseRadius + csetup.Radius + 1.5), 2);
+
+            // are we close enough?   If so, our work here is done.
+            if ((player.PhysicsData.Position.SquaredDistanceTo(targetWorldObject.PhysicsData.Position) <= arrivedRadius)) return true;
+
+            // Ok, not in range.   Save off the action for later, and start them on their way.
+            player.MoveToPosition = targetWorldObject.PhysicsData.Position;
+            player.ArrivedRadiusSquared = arrivedRadius;
+            player.BlockedGameAction = action;
+            player.OnAutonomousMove(targetWorldObject, movementType);
```

Comment:
yea now that you point it out - a combined two actions that should each be their own method.   I think I have to do all the math to see if I am in range.   I am setting a destination, a range and saving off the original action (pick something up, use something etc) and then starting on the way.     I guess the best way to do it is to have InRangeForAction return arrivedRadiusSquared - if non 0 then I can take the appropriate action.   So let it be written, so let it be done.